bridge-token-factory/tests/token_transfer.rs

```diff
@@ -669,3 +651,96 @@ async fn test_upgrade() {
         .unwrap();
     assert!(is_paused);
 }
+
+#[tokio::test]
+async fn test_attach_full_access_key() {
+    let (alice, factory, worker) = create_contract(FACTORY_WASM_PATH).await;
+
+    let token_account = Account::from_secret_key(
+        format!("{}.{}", DAI_ADDRESS, factory.id()).parse().unwrap(),
+        factory.as_account().secret_key().clone(),
+        &worker,
+    );
+
+    // Grant alice the `DAO` role
+    let result = factory
+        .call("acl_grant_role")
+        .args_json(json!({
+           "role": "DAO",
+           "account_id": alice.id().to_string()
+        }))
+        .max_gas()
+        .transact()
+        .await
+        .unwrap();
+    let result: bool = result.json().unwrap();
+    assert!(result);
+
+    assert!(alice
+        .call(factory.id(), "deploy_bridge_token")
+        .deposit(35 * ONE_NEAR)
+        .args(
+            json!({"address": DAI_ADDRESS.to_string()})
+                .to_string()
+                .into_bytes()
+        )
+        .max_gas()
+        .transact()
+        .await
+        .unwrap()
+        .is_success());
+
+    let token_account_id: String = factory
+        .view("get_bridge_token_account_id")
+        .args_json(json!({"address": DAI_ADDRESS.to_string()}))
+        .await
+        .unwrap()
+        .json()
+        .unwrap();
+
+    assert_eq!(token_account_id, token_account.id().to_string());
+
+    let ed_pk_str = "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp";
+
+    // Add new full access key to the factory contract
+    assert!(alice
+        .call(factory.id(), "attach_full_access_key")
+        .args_json(json!({ "public_key": ed_pk_str }))
+        .max_gas()
+        .transact()
+        .await
+        .unwrap()
+        .is_success());
+
+    let keys: Vec<String> = factory
+        .view_access_keys()
+        .await
+        .unwrap()
+        .iter()
+        .map(|info| info.public_key.to_string())
+        .collect();
+
+    // Check that the access key is added to the factory contract
+    assert!(keys.len() == 2 && keys.contains(&ed_pk_str.to_string()));
+
+    // Add new full access key to the bridge token contract
+    assert!(alice
+        .call(factory.id(), "attach_full_access_key_to_token")
+        .args_json(json!({ "public_key": ed_pk_str, "address": DAI_ADDRESS }))
+        .max_gas()
+        .transact()
+        .await
+        .unwrap()
+        .is_success());
+
+    let keys: Vec<String> = token_account
+        .view_access_keys()
+        .await
+        .unwrap()
+        .iter()
+        .map(|info| info.public_key.to_string())
+        .collect();
+
+    // Check that the access key is added to the bridge token contract
+    assert!(keys.len() == 1 && keys.contains(&ed_pk_str.to_string()));
```

Comment:
Optional, may be it can be a good idea also add test, which checks that account without access right cannot add full access key.